waddrmgr/db.go

```diff
@@ -862,14 +1225,108 @@ func existsAddress(tx walletdb.Tx, addressID []byte) bool {
 	return bucket.Get(addrHash[:]) != nil
 }
 
+// fetchAddrAccount returns the account to which the given address belongs to.
+// It looks up the account using the addracctidx index which maps the address
+// hash to it's corresponding account id.
+func fetchAddrAccount(tx walletdb.Tx, addressID []byte) (uint32, error) {
+	bucket := tx.RootBucket().Bucket(addrAcctIdxBucketName)
+
+	addrHash := fastsha256.Sum256(addressID)
+	val := bucket.Get(addrHash[:])
+	if val == nil {
+		str := "account not found"
+		return 0, managerError(ErrAccountNotFound, str, nil)
+	}
+	return binary.LittleEndian.Uint32(val), nil
+}
+
+// fetchAccountAddresses loads information about addresses of an account from the database.
+// The returned value is a slice address rows for each specific address type.
+// The caller should use type assertions to ascertain the types.
+func fetchAccountAddresses(tx walletdb.Tx, account uint32) ([]interface{}, error) {
+	bucket := tx.RootBucket().Bucket(addrBucketName)
+
+	accountBucket := tx.RootBucket().Bucket(addrAcctIdxBucketName).
+		Bucket(uint32ToBytes(account))
+	// if index bucket is missing the account, there hasn't been any address
+	// entries yet
+	if accountBucket == nil {
+		return []interface{}{}, nil
```

Comment:
This can be `return nil, nil` to be consistent with the rest of the codebase.  Any calling code doing either `len(addrs)` or `for _, addr := range addrs` will work properly.